front/src/routes/integration/all/tp-link/device-page/Device.jsx

```diff
@@ -0,0 +1,75 @@
+import { Text } from 'preact-i18n';
+import { Component } from 'preact';
+import cx from 'classnames';
+import { RequestStatus } from '../../../../../utils/consts';
+import DeviceForm from './DeviceForm';
+
+class TpLinkDeviceBox extends Component {
+  saveDevice = async () => {
+    this.setState({ loading: true });
+    this.setState({ saveError: null });
+    this.setState({ deleteError: null });
```

Comment:
Why don't you do all this in one setState?

```javascript
this.setState({
  loading: true,
  saveError: null,
  deleteError: null
});
```

setState is an expensive operation as it triggers a whole front pain. Even if preact has optimization to avoid painting the screen all the time, it's still better to do one setState instead of 3 :)

Reply:
Fixed (I'm not used to Frontend development but learning every day 😄)